src/main/webapp/dev/js/common/instructorFeedbackResults.js

```diff
@@ -151,45 +155,39 @@ function expandOrCollapsePanels(expandCollapseButton, panels) {
     const STRING_COLLAPSE = 'Collapse';
     const expandCollapseStudentsButton = 'a[id^="collapse-panels-button-section-"]';
     const expandCollapseTeamButton = 'a[id^="collapse-panels-button-team-"]';
-    const isButtonInExpandMode = $(expandCollapseButton).html().trim().startsWith(STRING_EXPAND);
     const isCollapsePanelsButton = !panels;
-    /*
-     * When collapsing expandCollapseButton, expandCollapseTeamButton and expandCollapseStudentButton
-     * should be in collapsed state, so that next time when we expand the main panel, we find
-     * the expandCollapseTeamButton and expandCollapseStudentButton in collapsed state.
-     */
-    if (isCollapsePanelsButton && !isButtonInExpandMode) {
-        if ($(expandCollapseStudentsButton).length
-            && $(expandCollapseStudentsButton).html().trim().startsWith(STRING_EXPAND)) {
-            replaceButtonHtmlAndTooltipText(expandCollapseStudentsButton, STRING_EXPAND, STRING_COLLAPSE);
-        }
-        if ($(expandCollapseTeamButton).length
-            && $(expandCollapseTeamButton).html().trim().startsWith(STRING_EXPAND)) {
-            replaceButtonHtmlAndTooltipText(expandCollapseTeamButton, STRING_EXPAND, STRING_COLLAPSE);
-        }
-    }
-    // When collapsing expandCollapseTeamButton, expandCollapseStudentButton should be in collapsed state.
-    if ($(expandCollapseStudentsButton).length && (expandCollapseButton === $(expandCollapseTeamButton))
-        && $(expandCollapseTeamButton).html().trim().startsWith(STRING_COLLAPSE)
-        && $(expandCollapseStudentsButton).html().trim().startsWith(STRING_EXPAND)) {
-        replaceButtonHtmlAndTooltipText(expandCollapseStudentsButton, STRING_EXPAND, STRING_COLLAPSE);
-    }
 
     // {@code panels} is not defined when {@code expandCollapseButton} is collapse panels button. We
     // need to define corresponding {@code targetPanels}.
     const targetPanels = panels || $('div.panel-collapse');
 
-    if (isButtonInExpandMode) {
+    if (isInExpandMode(expandCollapseButton)) {
+        // When expanding all panels via the expand panels button update the state for
+        // all expand/collapse buttons underneath (namely, team and student buttons).
+        if (isCollapsePanelsButton) {
+            if ($(expandCollapseStudentsButton).length
+                && isInExpandMode(expandCollapseStudentsButton)) {
+                replaceButtonHtmlAndTooltipText(expandCollapseStudentsButton, STRING_EXPAND, STRING_COLLAPSE);
+            }
+            if ($(expandCollapseTeamButton).length
+                && isInExpandMode(expandCollapseTeamButton)) {
+                replaceButtonHtmlAndTooltipText(expandCollapseTeamButton, STRING_EXPAND, STRING_COLLAPSE);
+            }
+        }
+        // When expanding expandCollapseTeamButton, expandCollapseStudentButton should be in collapsed state.
+        if ($(expandCollapseStudentsButton).length && (expandCollapseButton === $(expandCollapseTeamButton))
```

Comment:
For consistency with `if (isCollapsePanelsButton) {`, extract a constant `isCollapseTeamButton` and use
```javascript
} else if (isCollapseTeamButton
        && ...)
```